detekt-rules/src/test/kotlin/OrtPackageNamingTest.kt

```diff
@@ -0,0 +1,157 @@
+                """.trimIndent()
+            )
+
+            val findings = rule.lint(file.toPath())
+
+            findings should beEmpty()
+        }
+
+        "succeed for package in second-level module" {
+            val file = createFile(
+                dir = "clients/osv/src/main/kotlin",
+                content = """
+                    package org.ossreviewtoolkit.clients.osv
+                """.trimIndent()
+            )
+
+            val findings = rule.lint(file.toPath())
+
+            findings should beEmpty()
+        }
+
+        "succeed for sub-package in second-level module" {
+            val file = createFile(
+                dir = "clients/osv/src/main/kotlin/utils",
+                content = """
+                    package org.ossreviewtoolkit.clients.osv.utils
+                """.trimIndent()
+            )
+
+            val findings = rule.lint(file.toPath())
+
+            findings should beEmpty()
+        }
+
+        "ignore a dash in the module name" {
+            val file = createFile(
+                dir = "clients/clearly-defined/src/main/kotlin",
+                content = """
+                    package org.ossreviewtoolkit.clients.clearlydefined
+                """.trimIndent()
+            )
+
+            val findings = rule.lint(file.toPath())
+
+            findings should beEmpty()
+        }
+
+        "fail if an invalid package is used" {
+            val file = createFile(
+                dir = "model/src/main/kotlin",
+                content = """
+                    package com.example
+                """.trimIndent()
+            )
+
+            val findings = rule.lint(file.toPath())
+
+            findings should haveSize(1)
+        }
+
+        "fail if the package does not match the module" {
+            val file = createFile(
+                dir = "model/src/main/kotlin",
+                content = """
+                    package org.ossreviewtoolkit.cli
+                """.trimIndent()
+            )
+
+            val findings = rule.lint(file.toPath())
+
+            findings should haveSize(1)
+        }
+
+        "fail if the sub-package does not match the directory" {
+            val file = createFile(
+                dir = "model/src/main/kotlin/config",
+                content = """
+                    package org.ossreviewtoolkit.model.utils
+                """.trimIndent()
+            )
+
+            val findings = rule.lint(file.toPath())
+
+            findings should haveSize(1)
+        }
+    }
+})
+
+private fun TestConfiguration.createFile(dir: String, content: String): File {
+    // The file needs to be created in the below directory because detekt uses it to determine relative paths in tests.
+    val parent = File("build/classes/kotlin/test/$dir").safeMkdirs()
+
+    afterTest {
+        parent.safeDeleteRecursively(force = true)
+    }
+
+    val file = File(parent, "Test.kt")
+    file.writeText(content)
+    return file
```

Comment:
Nit: Could be `return File(parent, "Test.kt").apply { writeText(content) }`

Reply:
Done in #6573.